File: typescript/src/clients/system.ts
import { BaseClient } from "../client-base.js";
import type { RequestOptions } from "../types.js";

/** Health check response from the server. */
export interface HealthCheckResponse {
  readonly status: string;
}

/**
 * Client for the System API endpoints.
 * Provides methods to check server health status.
 *
 * No authentication is required for these endpoints.
 */
export class SystemClient extends BaseClient {
  /**
   * Check server health status.
   *
   * GET /health
   *
   * @param options - Optional request options (headers, signal, etc.)
   * @returns A promise that resolves with the health check response.
   */
  async healthCheck(options?: RequestOptions): Promise<HealthCheckResponse> {
    return this.get<HealthCheckResponse>("/health", undefined, options);
  }
}

File: python/src/opperai/clients/system.py
"""Opper SDK — System Client."""

from __future__ import annotations

from .._base_client import BaseClient
from ..types import HealthResponse, RequestOptions


class SystemClient:
    """Client for the System API endpoints."""

    def __init__(self, client: BaseClient) -> None:
        self._client = client

    def health(self, *, options: RequestOptions | None = None) -> HealthResponse:
        data = self._client._get("/health", options=options)
        return HealthResponse(status=(data or {}).get("status", ""))

    async def health_async(self, *, options: RequestOptions | None = None) -> HealthResponse:
        data = await self._client._get_async("/health", options=options)
        return HealthResponse(status=(data or {}).get("status", ""))
